salt/metalk8s/kubeadm/init/certs/ca.sls

```diff
@@ -0,0 +1,86 @@
+{%- from "metalk8s/map.jinja" import ca with context %}
+
+{%- set ca_server = salt['mine.get']('*', 'kubernetes_ca_server') %}
+
+{#- Check if we have no CA server or only current minion as CA #}
+{%- if not ca_server or ca_server.keys() == [grains['id']] %}
+
+include:
+  - .installed
+
+Create CA private key:
+  x509.private_key_managed:
+    - name: /etc/kubernetes/pki/ca.key
+    - bits: {{ ca.cert.private_key_size }}
+    - user: root
+    - group: root
+    - mode: 600
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - pkg: Install m2crypto
+
+Generate CA certificate:
+  x509.certificate_managed:
+    - name: /etc/kubernetes/pki/ca.crt
+    - signing_private_key: /etc/kubernetes/pki/ca.key
+    - CN: kubernetes
+    - keyUsage: "critical digitalSignature, keyEncipherment, keyCertSign"
+    - basicConstraints: "critical CA:true"
+    - days_valid: {{ ca.cert.days_valid }}
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+    - require:
+      - x509: Create CA private key
+
+# TODO: Find a better way to advetise CA server
+Advertise CA in the mine:
+  module.wait:
+    - mine.send:
+      - func: 'kubernetes_ca_server'
+      - mine_function: test.ping
```

Comment:
And, we need to put the CA cert in the kubeconfig files. Does it make sense if we put a base64 value of the CA instead of the ping result? Something like that: 
`salt-call hashutil.base64_encodefile /etc/kubernetes/pki/ca.crt`

Reply:
Oh yes we can. If it's needed yes I will do that

Reply:
Let's add it into a debt ticket.

Reply:
Or maybe could be part of your PR directly @ramanduh as it's needed for your PR.
So we can merge this one

Reply:
Yep, I'll do it in my ticket